### DISCUSSION TITLE
Cannot get arbitrary properties working on create-react-app

### DISCUSSION THREAD

**Discussion by tm-noelzubin** [tailwindlabs/tailwindcss#7036] [Help]
I installed tailwind on create-react-app starter. I cant seem to get arbitrary properties working. I'm running tailwind 3.0.8. Other tailwind css classes work fine. eg. although the class `top-[10px]` is applied it doesnt have any associated styled generated.

**Comment by NovusOpus** [comment]
Can you share a minimal reproduction that we can look at?

**Reply by tm-noelzubin** [reply]
I think it might be an issue with my configuration. It seems to be working on a new project

**Comment by simonswiss** [comment]
Hey - a verly likely issue is that file where you're using `top-[10px]` is not present in your `content` array:

https://tailwindcss.com/docs/content-configuration.

That, or you're trying to create a dynamic class (which won't work since the JIT engine needs to see the full class string in one piece to generate it:

```js
const offset = '10px'

<div className={`top-[${offset}]`}>
```

This won't work ☝️
```

**Reply by joaopedrosdmm** [reply]
```
const offset = '10px'

<div className={`top-[${offset}]`}>
```

So what is the alternative to this? styles?
And with version 3.023 do I need to npm install jit for dynamic styles to work?

**Reply by zslabs** [reply]
The workaround is to use full class names instead of partials like this. Think of the Tailwind CSS compiler as static, it can't read dynamically generated names.

**Reply by wongjn** [reply]
Set offset as a CSS variable. You could do something like:
```jsx
const offset = '10px';

<div className="top-[var(--offset)]" style={`--offset: ${offset}`} />
```

**Reply by theDanielJLewis** [reply]
This was the way I had to change it because the above style code wouldn't work.

```js
<div className="bg-[var(--offset)]" style={{ '--offset': offset }}>
```
